#!/bin/sh -e
## gcc4_buildfixes.dpatch by Henrique de Moraes Holschuh <hmh@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix FTBFS problems with gcc 4.0 on AMD64 (closes: #286353)
## DP: patch by Andreas Jochens <aj@andaco.de>

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad hplip/prnt/hpijs/ljcolor.cpp /tmp/dpep.EcmGCg/hplip/prnt/hpijs/ljcolor.cpp
--- hplip/prnt/hpijs/ljcolor.cpp	2005-01-21 00:19:51.000000000 -0200
+++ hplip/prnt/hpijs/ljcolor.cpp	2005-01-21 11:25:17.000000000 -0200
@@ -286,7 +286,7 @@
 
     if (thePrinter->IOMode.bStatus)
     {
-        sprintf (res, "@PJL JOB NAME = \"%d\"\015\012", (int) (thePrinter));
+        sprintf (res, "@PJL JOB NAME = \"%ld\"\015\012", (long) (thePrinter));
         err = thePrinter->Send ((const BYTE *) res, strlen (res));
         ERRCHECK;
     }
@@ -405,8 +405,8 @@
     DRIVER_ERROR    err = NO_ERROR;
     if (thePrinter->IOMode.bStatus)
     {
-        sprintf (szBuff, "\033E\033%%-12345X@PJL EOJ NAME = \"%d\"\015\012@PJL RESET\015\012",
-                 (int) (thePrinter));
+        sprintf (szBuff, "\033E\033%%-12345X@PJL EOJ NAME = \"%ld\"\015\012@PJL RESET\015\012",
+                 (long) (thePrinter));
         err = thePrinter->Send ((const BYTE *) szBuff, strlen (szBuff));
         ERRCHECK;
     }
@@ -585,7 +585,7 @@
         while (*tmpStr < '0' || *tmpStr > '9')
             tmpStr++;
         sscanf (tmpStr, "%d", &iErrorCode);
-        if (iErrorCode != (int) (this))
+        if (iErrorCode != (long) (this))
             return DISPLAY_PRINTING;
     }
 
diff -urNad hplip/prnt/hpijs/ljmono.cpp /tmp/dpep.EcmGCg/hplip/prnt/hpijs/ljmono.cpp
--- hplip/prnt/hpijs/ljmono.cpp	2005-01-21 00:19:51.000000000 -0200
+++ hplip/prnt/hpijs/ljmono.cpp	2005-01-21 11:25:17.000000000 -0200
@@ -185,7 +185,7 @@
 
     if (thePrinter->IOMode.bStatus)
     {
-        sprintf (res, "@PJL JOB NAME = \"%d\"\015\012", (int) (thePrinter));
+        sprintf (res, "@PJL JOB NAME = \"%ld\"\015\012", (long) (thePrinter));
         err = thePrinter->Send ((const BYTE *) res, strlen (res));
         ERRCHECK;
 
@@ -279,8 +279,8 @@
     DRIVER_ERROR    err = NO_ERROR;
     if (thePrinter->IOMode.bStatus)
     {
-        sprintf (szBuff, "\033E\033%%-12345X@PJL EOJ NAME = \"%d\"\015\012@PJL RESET\015\012",
-                (int) (thePrinter));
+        sprintf (szBuff, "\033E\033%%-12345X@PJL EOJ NAME = \"%ld\"\015\012@PJL RESET\015\012",
+                (long) (thePrinter));
         err = thePrinter->Send ((const BYTE *) szBuff, strlen (szBuff));
     }
 
@@ -350,7 +350,7 @@
         while (*tmpStr < '0' || *tmpStr > '9')
             tmpStr++;
         sscanf (tmpStr, "%d", &iErrorCode);
-        if (iErrorCode != (int) (this))
+        if (iErrorCode != (long) (this))
             return DISPLAY_PRINTING;
     }
 
